packages/integrations/image/src/vendor/squoosh/emscripten-utils.ts

```diff
@@ -29,3 +29,16 @@ export function instantiateEmscriptenWasm<T extends EmscriptenWasm.Module>(
 export function dirname(url: string) {
 	return url.substring(0, url.lastIndexOf('/'))
 }
+
+/**
+ * On certain serverless hosts, our ESM bundle is transpiled to CJS before being run, which means
+ * import.meta.url is undefined, so we'll fall back to __dirname in those cases
+ * We should be able to remove this once https://github.com/netlify/zip-it-and-ship-it/issues/750 is fixed
+ */
+export function getModuleURL(url: string | undefined): string {
+	if (!url) {
+		return pathToFileURL(__dirname).toString();
```

Comment:
The `__dirname` refers relative to this utils file. Would it work correctly if it's used but the consumer file later on?

I see some doing 
```js
scriptDirectory = dirname(getModuleURL(import.meta.url)) + '/'
```

which might have a different value depending on whether `import.meta.url` is supported.

Reply:
It's admittedly not ideal, but it does work because the __dirname call is only used in SSR where everything gets bundled 😅

If you have a better solution, I'll definitely take! Got lost in between the `__dirname`, `__filename` and `import.meta.url` while working on this so really I settled on the solution that actually did work 😵‍💫

Reply:
Ah I see. Didn't know this was bundled so it would always have the same value.

This trick sounds good to me then, though I wonder if it should be `__filename` instead since that reflects `import.meta.url`. If you've tested with `__dirname` and it works then I'll be fine by it too 👍